File: Stack/Valid Parenthesis.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> stack = new Stack<char>();
        for (int i = 0; i < s.Length; i++)
        {
            if (s[i] == '(' || s[i] == '[' || s[i] == '{') stack.Push(s[i]);
            else if (s[i] == ')' || s[i] == ']' || s[i] == '}')
            {
                if (stack.Count != 0)
                {
                    switch (s[i])
                    {
                        case ')':
                            {
                                if(stack.Peek() == '(') stack.Pop();
                                else return false;
                                break;
                            }
                        case ']':
                            {
                                if(stack.Peek() == '[') stack.Pop();
                                else return false;
                                break;
                            }
                        case '}':
                            {
                                if(stack.Peek() == '{') stack.Pop();
                                else return false;
                                break;
                            }
                    }
                }
                else return false;
            }
        }
        
        
        if (stack.Count == 0) return true;
        else return false;
    }
}

File: Stack/Valid Parenthesis.ts
function isValid(s: string): boolean {
    const leftBrackets = ['(', '{', '[']
    const arr = [];

    for (let i = 0; i < s.length; i++) {
        if (leftBrackets.includes(s[i])) {
            arr.push(s[i]);
        }
        else {
            const lastBracket = arr.pop();
            switch (lastBracket) {
                case '(' : 
                {
                    if (s[i] != ')') return false;
                    break;
                }
                case '{' :
                {
                    if (s[i] != '}') return false;
                    break;
                }
                case '[':
                {
                    if (s[i] != ']') return false;
                    break;
                }
                default: 
                {
                    return false;
                }
            }
        }
    }

    if (arr.length > 0) return false;

    return true;
};
